src/Glpi/Database/AbstractDatabase.php

```diff
@@ -1575,4 +1575,95 @@ public function notTzMigrated() :int
     {
         return 0;
     }
+
+    /**
+     * Builds a bulk insert statement
+     *
+     * @since 10.0.0
+     * @param \QueryExpression|string $table  Table name
+     * @param array                   $keys Array of column names
+     * @param array                   $params Array of arrays of values.
+     *                                    This will be replaced with an associative array for a PDO statement
+     * @return string
+     */
+    public function buildInsertBulk($table, array $columns, array &$params): string
```

Comment:
This seems to duplicate a lot of the buildInsert existing method. Also, I wonder if using a prepared statement to loop on each value would not be better than using multiple inserts

Reply:
Main benefit would be performance in upgrades and install. In my case, adding data during install using individual inserts took 4.3 seconds on average. Bulk inserting took 0.15 seconds. For most other things, individual inserts may be best in case the columns are different.